### ISSUE TITLE
Update Hydro Staff Domains

### ISSUE THREAD

**Issue by mripp** [NWFWMD-IT/Hydrologic-Monitoring-Project#124]
Since the question in Survey123 'Staff' allows the user to select multiple staff names there is a requirement to have no spaces in the domains. 

Fix: I recommend to add a underline to all domains
Example: Candice Keeney --> Candice_Keeney

In addition, I noticed there is an 'Other' domain option for staff but there isn't an extra field to record this like

**Comment by mmannion** [comment]
Added agenda item for August 28 team meeting to discuss:
- 'Other' choice
- Embedded spaces

**Comment by mmannion** [comment]
Per team meeting:
1. Add text field for "Other" choice
2. For `LocationVisit.Staf`
  - Remove _Hydro Staff_ domain
  - Keep multi-select list options defined in Survey itself

**Comment by mripp** [comment]
Communication between Sarah and Tech Support 11/8

_Hello Sarah,
 
Thank you for confirming that!  I have updates from my team and as the space character is categorized as a [special character](https://doc.arcgis.com/en/survey123/desktop/create-surveys/quickreferencecreatesurveys.htm#:~:text=not%20store%20values-,Special%20characters,-Question%20names%20and), we recommend and encourage posting an Idea to the [ArcGIS Survey123 Ideas](https://community.esri.com/t5/arcgis-survey123-ideas/idb-p/arcgis-survey123-ideas) Esri Community Board describing this functionality instead of logging a new Enhancement for it.  As we are not logging a new Enhancement request for this functionality at this time, I will proceed to close this Technical Support case. 
 
If you wanted, I am happy to meet over a phone call or screen share to talk in more detail about this functionality and the ArcGIS Ideas board and I am available today from 9:15 - 11:00 am and 3:00 - 4:30 pm EST or Monday - Thursday next week from 9:15 - 11:00 am EST and from 3:00 - 4:30 pm EST.  If these times don't work, please let me know a time that does!
 
Thank you, I hope that you have a great day!_

**Comment by mripp** [comment]
MLR will look into options to convert the underscores back into a space using an intermediate field in the survey

**Comment by mmannion** [comment]
Implemented data model changes:
- Removed _Hydro Staff_ domain
  - Value list will live in survey
  - Left list in data model diagram, for reference
- Added `LocationVisit.StaffComments` column
  - Enable in survey when choosing 'Other' value from `select_multiple` list of staff names
  - Column name is kinda weird, but consistent with other areas of model where 'Other' choice is paired with '*Comments' column (see #75)

**Comment by mripp** [comment]
@mmannion I do not see the new column `LocationVisit.StaffComments` in the feature layer. Will this field become available once you redeploy the database?

**Comment by mripp** [comment]
Implemented Javascript in the Survey in order to replace all  underscores in the Staff field to spaces
- Script is stored within Survey123 Connect under the `Scripts` tab
- After testing offline on my mobile device I confirmed that this JS functionality works while offline. More extensive testing of this field will need to happen during Testing Round 2

This is how it ends up looking in the database
![Image](https://github.com/NWFWMD-IT/Hydrologic-Monitoring-Project/assets/55856222/3ed679b4-27e1-4073-9ade-c93864e5c5ec)

**Comment by mmannion** [comment]
Yes; there's a question on the agenda for next meeting regarding when to deploy.

I can do it any time (including today, if you like) but it will refresh the data from source and I didn't want to overwrite any data you'd customized for testing (e.g. surveys collected, Locations created/modified to test edge cases).

Shall I deploy now, or would you like to discuss on Monday?

Note that I still need to add two new columns to `LocationVisit`:
- `HasSensor`
- `HasMeasuringPoint`

I can either deploy again when those are ready, or wait and deploy both changes at once. It's easy enough, so no trouble if it'd be helpful to have `StaffComments` now.

(I think I need to create a ticket for the `Has` columns; they're annotated in the data model diagram, but I don't think there's a task for them yet. I'll check now, create one if necessary, and implement those today. Sorry about that; I didn't realize those were outstanding, and I should have done them before working on the photo loader.)

**Comment by mripp** [comment]
Okay lets wait to deploy once all the fields have been created

**Comment by mmannion** [comment]
See #126 for `HasSensor/MeasuringPoint`. I'll have them ready shortly and will deploy this ticket and that to the production database today. I'll back up the existing database before overwriting it, in case we need it for demos/etc. on Monday.

**Comment by mripp** [comment]
I updated the survey including

- The field `StaffComments` can now be populated into the database
- The field `Staff` is now formatted into a comma separated list (without underscores and with a space after the comma)
